src/front/zsharp/mod.rs

```diff
@@ -305,6 +305,24 @@ impl<'ast> ZGen<'ast> {
                     Ok(uint_lit(DFL_T.modulus().significant_bits(), 32))
                 }
             }
+            "vector_add" => {
+                if args.len() != 2 {
+                    Err(format!(
+                        "Got {} args to EMBED/vector_add, expected 2",
+                        args.len()
+                    ))
+                } else if generics.len() != 1 {
+                    Err(format!(
+                        "Got {} generic args to EMBED/vector_add, expected 1",
+                        generics.len()
+                    ))
+                } else {
+                    assert!(args.iter().all(|t| matches!(t.type_(), Ty::Array(_, _))));
+                    let a = args.pop().unwrap();
+                    let b = args.pop().unwrap();
```

Comment:
```suggestion
                    let b = args.pop().unwrap();
                    let a = args.pop().unwrap();
```

`pop` takes from the end of the array, so you're reversing the order of arguments here.